### PR TITLE
Add --clean-per-run flag

### LINKED ISSUE #47: Option to clean between each run
I'm attempting to use `cargo hack test --feature-powerset` on a project, but given the sheer number of combinations this results in exceeding the available disk space on Github Actions with all the resulting test binaries.

If it were possible to run `cargo clean -p <project>` between each test run, that would delete the test binaries but leave the precompiled dependencies behind. It's probably not wanted as the default since it would slow down rerunning `cargo hack --feature-powerset` locally when you are able to cache all the results, but being an option would be very useful for CI.

### PR DESCRIPTION
**Pull request by taiki-e**
```text
--clean-per-run
    Remove artifacts for that package before running the command.
    
    If used this flag with --workspace, --each-feature, or --feature-powerset, 
    artifacts will be removed before each run.

    Note that dependencies artifacts will be preserved.
```

Closes #47 

cc @Nemo157

### CODE REVIEW

**Comment by taiki-e** [comment]
Given that `cargo clean -p` had a lot of bugs until recently, you probably need a beta or nightly compiler at this time for this to work as expected. (https://github.com/rust-lang/cargo/pull/8210)
However, it doesn't rely on unstable features, so it should be available at 1.45.

File: src/cli.rs [hunk 1]
```diff
@@ -54,6 +54,9 @@ const HELP: &[(&str, &str, &str, &[&str])] = &[
         "Skip passing --features flag to `cargo` if that feature does not exist in the package",
         &[],
     ),
+    ("", "--clean-per-run", "Remove artifacts for that package before running the command", &[
+        "If used this flag with --workspace, --each-feature, or --feature-powerset, artifacts will be removed before each run.",
```
Comment (Nemo157) [inline src/cli.rs:58]: You might want to mention that dependencies artifacts will be preserved.

Comment (taiki-e) [reply src/cli.rs:58]: Thanks, that certainly should be mentioned.

Comment (taiki-e) [reply src/cli.rs:59]: Done in c93bc5d

File: src/package.rs [hunk 2]
```diff
@@ -200,6 +204,20 @@ fn exec_cargo(
...
+fn cargo_clean(cargo: &OsStr, args: &Args, package: &Package<'_>) -> Result<()> {
+    let mut line = ProcessBuilder::new(cargo, args.verbose);
+    line.arg("clean");
+    line.arg("--package");
+    line.arg(&package.name);
+
+    if args.verbose {
+        // running `cargo clean --package <package>`
+        info!(args.color, "running {}", line);
+    }
```
Comment (taiki-e) [inline src/package.rs:216]: It doesn't display by default, but it's probably fine. (displayed when you pass the `-v` flag.)

**Pull request merged**